### PR TITLE
add device_integration github issue template

### PR DESCRIPTION
**Pull request by maurerle**
this *should* add a device_integration github issue template.
I don't know if there is an easy way to test this.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/device_integration.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+<!--
+
+Please name your PR like "target: add device name".
+For example, to add Xiaomi Mi Router 4A (Gigabit Edition) name the PR:
+`ramips-mt7621: add Xiaomi Mi Router 4A`
```
Comment (herbetom) [inline .github/PULL_REQUEST_TEMPLATE/device_integration.md:5]: I would prefer a less shitty device as an example, also the syntax in the past was mostly "add support for" AFAIK. Why did you leave out the "(Gigabit Edition)"?

```suggestion
Please name your PR like "$TARGET: add support for $DEVICE_NAME".
For example, to add Xiaomi Mi Router 4A (Gigabit Edition) name the PR:
`ramips-mt7621: add support for Xiaomi Mi Router 4A (Gigabit Edition)`
```

File: .github/PULL_REQUEST_TEMPLATE/device_integration.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
+<!--
+
```
Comment (herbetom) [inline .github/PULL_REQUEST_TEMPLATE/device_integration.md:2]:
```suggestion
All pull requests should target master. A backport can be made afterwards and is welcomed (as long as it is easy to realize). Next is not a valid target.

```

File: .github/PULL_REQUEST_TEMPLATE/device_integration.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+<!-- 
+if wanted you can add infos about the used configuration for testing here
+-->
+
+**Gluon Version:**
+**Site Configuration:**
+
+<!--
+don't forget to add your device also to `docs/user/supported_devices.rst`
+-->
```
Comment (herbetom) [inline .github/PULL_REQUEST_TEMPLATE/device_integration.md:68]:
```suggestion
```
duplicate

File: .github/PULL_REQUEST_TEMPLATE/device_integration.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+- Outdoor devices only:
+  - [ ] Added board name to `is_outdoor_device` function in `package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua`
+- Docs:
+  - [ ] Added Device to `docs/user/supported_devices.rst`
+
+<!-- 
+if wanted you can add infos about the used configuration for testing here
+-->
+
+**Gluon Version:**
```
Comment (herbetom) [inline .github/PULL_REQUEST_TEMPLATE/device_integration.md:63]: Mhm, what's the point? The tests should be done with the state of the Commit.


```suggestion
```

**Review by herbetom — CHANGES_REQUESTED** [review]
testing is possible for example via 

https://github.com/maurerle/gluon/compare/feature/device_integration...herbetom:gluon:add_linksys-ea8300

Also this template shouldn't be the only one. Not everything is device support.

File: .github/PULL_REQUEST_TEMPLATE/device_integration.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+Next fill out the following device integration checklist and make notes if something unexpected happens
+
+You can use the "Preview" tab to check how your issue is going to look
+before you actually send it in.
+
+Thank you for taking the time to integrate a new device into the Gluon project.
+
+-->
+
+### Device Integration Checklist
```
Comment (herbetom) [inline .github/PULL_REQUEST_TEMPLATE/device_integration.md:16]:
```suggestion
```

probably not necessary

File: .github/PULL_REQUEST_TEMPLATE/device_integration.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    - [ ] Should display config mode blink sequence 
+          (https://gluon.readthedocs.io/en/latest/features/configmode.html)
+  - Radio LEDs
+    - [ ] Should map to their respective radio
+    - [ ] Should show activity
+  - Switch port LEDs
+    - [ ] Should map to their respective port (or switch, if only one led present) 
+    - [ ] Should show link state and activity
+- Outdoor devices only:
+  - [ ] Added board name to `is_outdoor_device` function in `package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua`
```
Comment (herbetom) [inline .github/PULL_REQUEST_TEMPLATE/device_integration.md:55]:
```suggestion
  - [ ] Added board name to `is_outdoor_device` function in `package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua`
 - Cellular devices only:
  - [ ] Added board name to `is_cellular_device` function in `package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua`
  - [ ] Added board name with modem setup function `setup_ncm_qmi` to `package/gluon-core/luasrc/lib/gluon/upgrade/250-cellular`
```

**Comment by maurerle** [comment]
So I found out, that it works having a `.github/PULL_REQUEST_TEMPLATE.md` but not of if you are using `.github/PULL_REQUEST_TEMPLATE/my_template.md`

any suggestions?

**Comment by herbetom** [comment]
Yeah, seems like that multiple Templates are only somewhat supported. It's possible to use those templates if you use a parameter in your URL to create the PR:

https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/using-query-parameters-to-create-a-pull-request

Maybe it's possible to link from `github/pull_request_template.md` to the correct template? But probably not easily and elegantly possible.

Astonishing that multiple Pull Request Templates aren't a thing.

**Comment by maurerle** [comment]
When pushing a branch, the git remote respondes with:

```
remote: Create a pull request for 'feature/test' on GitHub by visiting:
remote:      https://github.com/maurerle/gluon/pull/new/feature/device_integration
```

which doesn't give such an option too.

This works:
https://github.com/maurerle/gluon/compare/feature/device_integration...herbetom:gluon:add_linksys-ea8300?quick_pull=1&template=device_integration_template.md

But currently, the feature is not very useful.
We could merge it either, and hope that it works some day with some kind of dropdown menu.. But then we can reopen this too

**Pull request closed without merging**